compiler/rustc_codegen_llvm/src/gotoc/cbmc/goto_program/expr.rs

```diff
@@ -416,6 +416,32 @@ impl Expr {
         }
     }
 
+    // Attempts automatic casting of arguments to type of function parameters
+    pub fn cast_function_arguments(
+        function: &Expr,
+        mut arguments: Vec<Expr>,
+        mm: &MachineModel,
+    ) -> Vec<Expr> {
+        let parameters = function.typ().parameters().unwrap();
+        assert!(arguments.len() >= parameters.len());
+        let mut rval: Vec<_> = parameters
+            .iter()
+            .map(|parameter| {
+                let argument = arguments.remove(0);
+                if parameter.typ() == argument.typ() {
+                    argument
+                } else {
+                    assert!(argument.typ().is_equal_on_machine(parameter.typ(), mm));
+                    argument.cast_to(parameter.typ().clone())
+                }
```

Comment:
This could be factored out into an independently useful function

Reply:
Fixed